iterator.go

```diff
@@ -0,0 +1,41 @@
+//go:build go1.22 && goexperiment.rangefunc
+// +build go1.22,goexperiment.rangefunc
+
+package gitlab
+
+import (
+	"iter"
+)
+
+// PageIterator is an EXPERIMENTAL iterator as defined in the "rangefunc" experiment for go 1.22.
+// See https://go.dev/wiki/RangefuncExperiment for more details.
+//
+// It can be used as:
+//
+//	for user, err := range gitlab.PageIterator(gl.Users.List, nil) {
+//		if err != nil {
+//			// handle error
+//		}
+//		// process individual user
+//	}
+func PageIterator[O, T any](f Paginatable[O, T], opt *O) iter.Seq2[*T, error] {
+	return func(yield func(*T, error) bool) {
+		nextLink := ""
+		for {
+			page, resp, err := f(opt, WithKeysetPaginationParameters(nextLink))
```

Comment:
The function `f` accepts a variable number of `RequestOptionFunc` arguments, but in this call here, it passes none of them.

Reply:
Maybe something like that
```go
//	for user, err := range gitlab.PageIterator(gl.Users.List, nil, nil) {
//		if err != nil {
//			// handle error
//		}
//		// process individual user
//	}
func PageIterator[O, T any](f Paginatable[O, T], opt *O, optionFunc ...gitlab.RequestOptionFunc) iter.Seq2[*T, error] {
	return func(yield func(*T, error) bool) {
		nextLink := ""
		for {
			page, resp, err := f(opt, append(optionFunc, gitlab.WithKeysetPaginationParameters(nextLink))...)
```

Reply:
ah yes, good point! Thankfully a quick fix. PTAL